lib/portlayer/exec/exec.go

```diff
@@ -76,12 +80,47 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 			Config.ResourcePool = o.ResourcePool
 		case *object.ResourcePool:
 			Config.ResourcePool = o
+			Config.VMGroupName = o.Name()
+			rp := compute.NewResourcePool(ctx, sess, cr)
+			Config.Cluster, err = rp.GetCluster(ctx)
+			if err != nil {
+				err = fmt.Errorf("could not get cluster from resource pool: %s", err)
```

Comment:
this is a horrible way to handle an error. it is here by initial design, but I would consider making it better.

Reply:
Curious how you'd improve it without significant increasing the scope of the PR. If you've a good suggestion, let me know.

Reply:
Easy.
```Go
subInitFunc := func() error {
    // logic with all returns
}
initializer.once.Do(func() {
    initializer.err = subInitFunc()
    if initalizer.err != nil {
        log.Errorf(initializer.err)
    }
})
```

In this case function becomes testable if moved out of the Init function and can be executed many times instead of just once.

Reply:
Think I'd put that in #6943 if anywhere in an existing PR.